wallet/txrules/poolfees.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2016 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+package txrules
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+
+	"github.com/decred/dcrd/blockchain"
+	"github.com/decred/dcrd/chaincfg"
+	"github.com/decred/dcrutil"
+)
+
+// maxPoolFeeRate is the maximum value of the pool fee
+const maxPoolFeeRate = 10000.0
+
+// IsValidPoolFeeRate tests to see if a pool fee is a valid percentage from
+// 0.01% to 100.00%.
+func IsValidPoolFeeRate(feeRate float64) error {
+	poolFeeRateTest := feeRate * 100
+	poolFeeRateTest = math.Floor(poolFeeRateTest)
+	if poolFeeRateTest <= 0.0 || poolFeeRateTest > maxPoolFeeRate {
+		return fmt.Errorf("invalid pool fee %v, pool fee "+
+			"must be between 0.01 and 100.00", feeRate)
+	}
+
+	return nil
+}
+
+// StakePoolTicketFee determines the stake pool ticket fee for a given ticket
+// from the passed percentage. Pool fee as a percentage is truncated from 0.01%
+// to 100.00%. This all must be done with integers, so bear with the big.Int
+// usage below.
+//
+// See the included doc.go of this package for more information about the
+// calculation of this fee.
+func StakePoolTicketFee(stakeDiff dcrutil.Amount, fee dcrutil.Amount,
```

Comment:
is fee the relay fee (looks like it based on the test vectors)? can it be renamed to relayFee?